Travis: Integration tests and setup script

aml2732 opened the pull request:
A setup script called prepIntegrationTests.sh which sets up environment variables and actions that the integration tests need to run.

Minimal modifications to tests to allow option to use -insecure creds when running using test credentials.

Inclusion of integration-related script in .travis.yml

Also, Docs proofed by : Josh K

jthomas commented on test/integration/prepIntegrationTests.sh:
```diff
@@ -0,0 +1,122 @@
+fi
+
+# Assert NODEJS6/NPM3 is default
+if [[ ! $(node --version) =~ v6.[0-9]*.[0-9]* ]]; then
```
We should change the version check to support anything from 6 and above. Node.js 8 becomes the LTS version in a few months. https://github.com/nodejs/LTS

aml2732 replied:
oh man. I'll have to look into how to do that.

jthomas commented on test/integration/actions.test.js:
```diff
@@ -112,14 +116,14 @@ test('create, get and delete with parameters an action', t => {
 })
 
-test('invoke action with fully-qualified name', t => {
+tempTest('invoke action with fully-qualified name', t => {
```
Has this been skipped because that action is not installed on the integration version? I'd prefer to work out a way to make this test work rather than skipping. We could install the `whisk.system/utils` package during setup on the integration host or re-write the test to invoke another available action using the fully qualified name. The former seems preferable?

aml2732 replied:
Feeds is skipped because alarm isn't installed/included by default during the openwhisk setup process. 

Justin informed me this would be difficult to do and suggested - to not do it.
For action this makes use of whisk.system/utils - which - I found out whisk.system stuff doesn't seem to be present when travis builds. (I was using the script copied from catalog - so this is probably some difference between my personal setup and travis script setup I don't fully understand). 

This should only be skipped when running  the tests with the -i flag. So that was the compromise I did. 

If you want me to not skip them, I need someone to direct me to setup-instructions/part-of-code/etc where whisk.system stuff is being made available. 

And even with that, alarms might still be a problem based on feedback I've gotten from Justin.

Not sure how to proceed. Will wait for your reply.

csantanapr replied:
catalog can be deploy using postdeploy in ansible.
alarm, kafka, and cloudant can't be deploy right now. but things like `/whisk.system/utils/echo` should be able to deploy

jthomas commented on test/integration/feeds.test.js:
```diff
@@ -19,15 +21,16 @@ envParams.forEach(key => {
+var tempTest = Utils.getInsecureFlag() ? test.skip : test;
 
-test('create and delete a feed', t => {
+tempTest('create and delete a feed', t => {
```
Same as above about not skipping tests.

jthomas commented:
Hey @aml2732, I've reviewed the incoming changes and added comments above.

Overall, this is looking great, nice work! 👏.
Looking forward to having integration tests working for the client library.

csantanapr commented on tools/travis/integration.sh:
```diff
@@ -0,0 +1,50 @@
+
+$ANSIBLE_CMD wipe.yml
+$ANSIBLE_CMD openwhisk.yml
+$ANSIBLE_CMD apigateway.yml
```
can deploy `postdeploy.yml` to make catalog packages like `/whisk.system/utils/` available

csantanapr commented on tools/travis/integration.sh:
```diff
@@ -0,0 +1,50 @@
+# Install OpenWhisk
+cd $WHISKDIR/ansible
+
+ANSIBLE_CMD="ansible-playbook -i environments/local"
```
I think we can use the images from docker hub making travis builds super fast by not needing to build the images need to pass the option `-e docker_image_prefix=openwhisk`

csantanapr commented on tools/travis/integration.sh:
```diff
@@ -0,0 +1,50 @@
+
+cd $WHISKDIR
+
+./gradlew distDocker
```
No need to build anything, I don't think we need the CLI to build

csantanapr commented on tools/travis/integration.sh:
```diff
@@ -0,0 +1,50 @@
+export OPENWHISK_HOME=$WHISKDIR
+
+# Set up CLI tool used by prereq script
+sudo cp $WHISKDIR/bin/wsk /usr/bin/wsk
```
Hum there is no need for the CLI, the integration tests should use env variables, in case you want to use `$HOME/.wskprops` I would say to put the values directly and avoid building the CLI.

aml2732 commented:
Updated with Carlos' changes from here: https://github.com/csantanapr/incubator-openwhisk-client-js/commits/angela/issue_2  to ^node6+, 7min builds, etc.
Also removed some extraneous stuff from prepIntegrationTests.sh

Only alarms test is skipped.
CLI is needed, as discussed with Carlos, until a later time.
Tested prepIntegrationTest.sh in both travis with guest creds node6/npm3, and locally using bluemix credentials node8/npm5.
